router/router.go

```diff
@@ -131,6 +134,15 @@ func (r *Router) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	h := r.h
 	r.l.RUnlock()
 
+	if r.sessionToken != "" {
```

Comment:
note for the future: This is probably better served as a http middleware rather than being in the handler itself. Typical patter of handlers doing handle-y stuff and http middlewares (through go chi or whatnot) handling auth, compression (note to self: we might want to enable gzip in the future, free latency boost), etc.

By doing so the entire auth stuff could live in cmd rather than being in core (and in the future, if we wanted and I'm not sure we will, we could have `run --no-auth`, `--auth-basic <token>`, etc without having to pollute the Router)